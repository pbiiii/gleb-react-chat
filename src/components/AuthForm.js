import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import UserFormInputType from '../types/UserFormInputType';

const AuthForm = ({
    username, password, repeatedPassword, onChange, onSubmit, isRegisterForm,
}) => (
    <form onSubmit={onSubmit}>
        <TextField
            required
            fullWidth
            label="Username"
            placeholder="Type your username..."
            type="text"
            name="username"
            margin="normal"
            autoComplete="username"
            value={username.value}
            onChange={onChange}
            error={!username.isValid}
        />
        <TextField
            required
            fullWidth
            label="Password"
            placeholder="Type your password..."
            type="password"
            name="password"
            margin="normal"
            autoComplete="current-password"
            value={password.value}
            onChange={onChange}
            error={!password.isValid}
        />
        {isRegisterForm && (
            <TextField
                required
                fullWidth
                label="Repeat password"
                placeholder="Repeat your password..."
                type="password"
                name="repeatedPassword"
                margin="normal"
                autoComplete="current-password"
                value={repeatedPassword.value}
                onChange={onChange}
                error={!repeatedPassword.isValid}
            />
        )}
        <Button fullWidth variant="contained" type="submit" color="primary">
            {isRegisterForm ? 'Register' : 'Login'}
        </Button>
    </form>
);

AuthForm.propTypes = {
    username: PropTypes.shape(UserFormInputType).isRequired,
    password: PropTypes.shape(UserFormInputType).isRequired,
    repeatedPassword: PropTypes.shape(UserFormInputType).isRequired,
    onChange: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    isRegisterForm: PropTypes.bool,
};

AuthForm.defaultProps = {
    isRegisterForm: false,
};

export default AuthForm;
