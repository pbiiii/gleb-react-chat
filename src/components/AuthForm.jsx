import React from 'react'
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

export const AuthForm = ({username, password, repeatedPassword, onChange, onSubmit, isRegisterForm}) => (
    <form onSubmit={onSubmit}>
        <TextField
            required
            fullWidth
            label="Username"
            placeholder="Type your username..."
            type="text"
            name="username"
            margin="normal"
            autoComplete="username"
            value={username.value}
            onChange={onChange}
            error={!username.isValid}
        />
        <TextField
            required
            fullWidth
            label="Password"
            placeholder="Type your password..."
            type="password"
            name="password"
            margin="normal"
            autoComplete="current-password"
            value={password.value}
            onChange={onChange}
            error={!password.isValid}
        />
        {isRegisterForm &&
        <TextField
            required
            fullWidth
            label="Repeat password"
            placeholder="Repeat your password..."
            type="password"
            name="repeatedPassword"
            margin="normal"
            autoComplete="current-password"
            value={repeatedPassword.value}
            onChange={onChange}
            error={!repeatedPassword.isValid}
        />
        }
        <Button
            fullWidth
            variant="contained"
            type="submit"
            color="primary"
        >
            {isRegisterForm ?
                'Register' :
                'Login'
            }
        </Button>
    </form>
)