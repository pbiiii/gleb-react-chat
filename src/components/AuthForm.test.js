/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AuthForm from './AuthForm';

const mockProps = {
    username: { value: 'username', valid: true },
    password: { value: 'password', valid: true },
    repeatedPassword: { value: 'password', valid: true },
    onChange: () => {},
    onSubmit: () => {},
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AuthForm {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<AuthForm {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
