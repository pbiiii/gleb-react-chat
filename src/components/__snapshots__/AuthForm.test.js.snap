// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
  >
    <label
      className="MuiFormLabel-root-104 MuiFormLabel-error-107 MuiInputLabel-error-96 MuiFormLabel-filled-108 MuiFormLabel-required-109 MuiInputLabel-required-97 MuiInputLabel-root-93 MuiInputLabel-formControl-98 MuiInputLabel-animated-101 MuiInputLabel-shrink-100"
      data-shrink={true}
    >
      Username
      <span
        className="MuiFormLabel-asterisk-110 MuiFormLabel-error-107 MuiInputLabel-error-96"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-124 MuiInput-root-111 MuiInput-underline-115 MuiInputBase-error-130 MuiInput-error-116 MuiInputBase-fullWidth-133 MuiInput-fullWidth-118 MuiInputBase-formControl-125 MuiInput-formControl-112"
      onClick={[Function]}
    >
      <input
        aria-invalid={true}
        autoComplete="username"
        className="MuiInputBase-input-134 MuiInput-input-119"
        disabled={false}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your username..."
        required={true}
        type="text"
        value="username"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-89 MuiFormControl-marginNormal-90 MuiFormControl-fullWidth-92"
  >
    <label
      className="MuiFormLabel-root-104 MuiFormLabel-error-107 MuiInputLabel-error-96 MuiFormLabel-filled-108 MuiFormLabel-required-109 MuiInputLabel-required-97 MuiInputLabel-root-93 MuiInputLabel-formControl-98 MuiInputLabel-animated-101 MuiInputLabel-shrink-100"
      data-shrink={true}
    >
      Password
      <span
        className="MuiFormLabel-asterisk-110 MuiFormLabel-error-107 MuiInputLabel-error-96"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-124 MuiInput-root-111 MuiInput-underline-115 MuiInputBase-error-130 MuiInput-error-116 MuiInputBase-fullWidth-133 MuiInput-fullWidth-118 MuiInputBase-formControl-125 MuiInput-formControl-112"
      onClick={[Function]}
    >
      <input
        aria-invalid={true}
        autoComplete="current-password"
        className="MuiInputBase-input-134 MuiInput-input-119 MuiInputBase-inputType-137 MuiInput-inputType-122"
        disabled={false}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your password..."
        required={true}
        type="password"
        value="password"
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-167 MuiButton-root-141 MuiButton-contained-152 MuiButton-containedPrimary-153 MuiButton-raised-155 MuiButton-raisedPrimary-156 MuiButton-fullWidth-166"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-142"
    >
      Login
    </span>
    <span
      className="MuiTouchRipple-root-170"
    />
  </button>
</form>
`;
